src/Windows/Host/Client/Impl/BrokerServices/LocalStaticFileServer.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.IO;
+using Microsoft.Common.Core.Logging;
+
+namespace Microsoft.R.Host.Client.BrokerServices {
+    public class LocalStaticFileServer : StaticFileServerBase {
+        private static LocalStaticFileServer _server;
+        private static object _serverLock = new object();
+
+        public LocalStaticFileServer(IFileSystem fs, IActionLog log, IConsole console) : base(fs, log, console) {
+        }
+
+        public async Task<string> HandleUrlAsync(string urlStr, CancellationToken ct = default(CancellationToken)) {
+            await InitializeAsync(ct);
+
+            Log.WriteLine(LogVerbosity.Minimal, MessageCategory.General, Resources.Info_StaticFileServerStarted.FormatInvariant(Listener.Prefixes.FirstOrDefault()));
+            Console.WriteLine(Resources.Info_StaticFileServerStarted.FormatInvariant(Listener.Prefixes.FirstOrDefault()));
+
+            UriBuilder ub = new UriBuilder(urlStr);
+            return GetFileServerPath(ub.Path);
+        }
+
+        public static Task<string> CreateAsync(string url, IFileSystem fs, IActionLog log, IConsole console, CancellationToken ct = default(CancellationToken)) {
```

Comment:
The method is misleading - from the name and signature, it would seem that it creates a new instance every time, but in practice it's a singleton accessor (and worse yet, it will silently ignore the arguments if singleton is already there, potentially giving you an instance with wrong services... at the very least, this might break tests).

I think it's better to manage the lifetime of that instance in the place where it is used, i.e. `RemotingWebServer`.